cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/TaskWorkerServiceLauncher.java

```diff
@@ -269,6 +269,17 @@ public void run() {
             }
           }
 
+          if (cConf.getBoolean(TaskWorker.TASK_WORKER_PROBE_ENABLED)) {
+            if (twillPreparer instanceof ExtendedTwillPreparer) {
+              String probeKeyPrefix = String.format("%s%s.",
+                  TaskWorker.TASK_WORKER_PROBE_PREFIX,
+                  cConf.get(TaskWorker.TASK_WORKER_PROBE_ENV));
+              twillPreparer = ((ExtendedTwillPreparer) twillPreparer)
+                  .addProbes(TaskWorkerTwillRunnable.class.getSimpleName(),
+                      cConf.getPropsWithPrefix(probeKeyPrefix));
```

Comment:
At `ExtendedTwillPreparer` send only the properties prefixed with `task.worker.probe`.

Environment level handling should be done in the implementation.